pkg/datapath/iptables/iptables.go

```diff
@@ -1016,7 +1016,7 @@ func (m *IptablesManager) installMasqueradeRules(prog, ifName, localDeliveryInte
 			m.waitArgs,
 			"-t", "nat",
 			"-A", ciliumPostNatChain,
-			"!", "-s", hostMasqueradeIP,
+			"!", "-s", allocRange,
```

Comment:
Had to think about it for a bit, but LGTM. The comment above needs updating though I think, "if the source is not the internal IP" is the part that's changing here. It also mentions tunnel and non-tunnel modes, but this rule installation logic is only handled in the tunneling case.